x/authority/types/authorizations_test.go

```diff
@@ -0,0 +1,306 @@
+package types_test
+
+import (
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/x/authority/types"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	lightclienttypes "github.com/zeta-chain/zetacore/x/lightclient/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func TestAuthorizationList_SetAuthorizations(t *testing.T) {
+	tt := []struct {
+		name             string
+		oldList          types.AuthorizationList
+		addAuthorization types.Authorization
+		expectedList     types.AuthorizationList
+	}{
+		{
```

Comment:
Can we add a test when `oldList` is empty?
Also might be relevant to add one where we modify an element in the middle of the list